src/main/java/org/opensearch/knn/index/query/ExactSearcher.java

```diff
@@ -68,26 +68,25 @@ private Map<Integer, Float> scoreAllDocs(KNNIterator iterator) throws IOExceptio
     }
 
     private Map<Integer, Float> searchTopK(KNNIterator iterator, int k) throws IOException {
-        // Creating min heap and init with MAX DocID and Score as -INF.
         final HitQueue queue = new HitQueue(k, true);
-        ScoreDoc topDoc = queue.top();
         final Map<Integer, Float> docToScore = new HashMap<>();
         int docId;
+
         while ((docId = iterator.nextDoc()) != DocIdSetIterator.NO_MORE_DOCS) {
-            if (iterator.score() > topDoc.score) {
-                topDoc.score = iterator.score();
-                topDoc.doc = docId;
-                // As the HitQueue is min heap, updating top will bring the doc with -INF score or worst score we
-                // have seen till now on top.
-                topDoc = queue.updateTop();
+            float score = iterator.score();
+
+            if (score < 0) {
```

Comment:
is the score conversion happening? because inner product can give negative scores

Reply:
Also, I didn't understand the logic on why score < 0, we should continue.